test/res_man_test.rb

```diff
@@ -0,0 +1,223 @@
+            @start_date, 
+            @end_date
+         )
+      end
+
+      it "finds reservation with room number match" do
+         matching_reservations = @phillip.reservations_by_room(
+            @room_num,
+            @start_date,
+            @end_date
+         )
+         expect(matching_reservations.length).must_equal 1
+         expect(matching_reservations[0]).must_equal @reservation
+         expect(matching_reservations[0].room.room_num).must_equal @room_num
+      end
+
+      it "finds no reservation if no room number match" do
+         matching_reservations = @phillip.reservations_by_room(
+            999,
+            @start_date,
+            @end_date
+         )
+         expect(matching_reservations.length).must_equal 0
+      end
+
+      it "finds no reservation if no date match" do
+         matching_reservations = @phillip.reservations_by_room(
+            @room_num,
+            Date.new(2019, 03, 10),
+            Date.new(2019, 03, 20),
+         )
+         expect(matching_reservations.length).must_equal 0
+      end
+   end
+
+   describe "reservations_by_date method" do
+      before do
+         @check_in = Date.new(2020, 3, 10)
+         @check_out = Date.new(2020, 3, 11)
+         @reservation = @phillip.make_reservation(
+            @check_in, 
+            @check_out
+         )
+      end
+
+      it "can retrieve a reservation given a valid date" do
+         matching_reservations = @phillip.reservations_by_date(@check_in)
+         
+         expect(matching_reservations.length).must_equal 1
+         expect(matching_reservations[0]).must_equal @reservation
+      end
+
+      it "will not retrieve any reservations for a non-matching date" do
+         matching_reservations = @phillip.reservations_by_date(
+            Date.new(2019, 03, 20)
+         )
+         
+         expect(matching_reservations.length).must_equal 0
+      end
+   end 
+
+   describe "view_by_vacancy method" do
+      it "retrieves all rooms for a range with no bookings" do
+         check_in = Date.new(2020, 05, 1)
+         check_out = Date.new(2020, 05, 1)
+
+         vacancies = @phillip.view_by_vacancy(check_in, check_out)
+
+         expect(vacancies.length).must_equal 20
+         vacancies.each do |room|
+            expect(room).must_be_instance_of Stayappy::Room
+         end
+      end
+
+      it "excludes a reserved room when a booking is matched" do
+         @check_in = Date.new(2020, 3, 10)
+         @check_out = Date.new(2020, 3, 11)
+         @reservation = @phillip.make_reservation(
+            @check_in, 
+            @check_out
+         )
+
+         vacancies = @phillip.view_by_vacancy(@check_in, @check_out)
+
+         expect(vacancies.length).must_equal 19
+         vacancies.each do |room|
+            expect(room.room_num).wont_equal @reservation.room.room_num
+         end
+      end
+
+      it "excludes a reserved room when a booking is matched" do
```

Comment:
It looks like this test is specific to excluding a room when it is included in a block. Nice work including this test! Make sure the string of the `it` block describes precisely this test case.